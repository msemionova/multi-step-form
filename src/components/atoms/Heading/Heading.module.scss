.heading {
  h1 {
    font: var(--typography-desktop-heading);
    margin-bottom: 1.6rem;
  }
  p {
    font: var(--typography-subheading);
    color: var(--color-gray-text);
  }
}
