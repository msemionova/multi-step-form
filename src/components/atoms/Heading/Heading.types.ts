export type HeadingProps = {
  title: string;
  subtitle?: string;
}
