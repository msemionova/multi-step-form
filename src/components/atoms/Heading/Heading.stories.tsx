import type { Meta, StoryObj } from '@storybook/react';

import Heading from './Heading';

const meta: Meta<typeof Heading> = {
  component: Heading,
};

export default meta;
type Story = StoryObj<typeof Heading>;

export const Default: Story = {
  args: {
    title: 'Title',
    subtitle: 'Subtitle',
  }
};
