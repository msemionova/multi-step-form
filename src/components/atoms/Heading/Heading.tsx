import styles from './Heading.module.scss';
import { HeadingProps } from './Heading.types';

export default function Heading({ title, subtitle }: HeadingProps) {
  return (
    <div className={styles.heading}>
      <h1>{title}</h1>
      <p>{subtitle}</p>
    </div>
  );
}
