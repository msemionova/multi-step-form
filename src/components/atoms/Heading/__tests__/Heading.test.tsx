import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import Heading from '../Heading';

describe('<Heading />', () => {
  const title = 'Title';
  const subtitle = 'Subtitle';

  it('renders correctly', () => {
    const tree = renderer
      .create(<Heading title={title} subtitle={subtitle} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
