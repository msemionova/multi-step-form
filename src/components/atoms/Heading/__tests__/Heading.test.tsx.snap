// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> renders correctly 1`] = `
<div
  className="heading"
>
  <h1>
    Title
  </h1>
  <p>
    Subtitle
  </p>
</div>
`;
